### ISSUE TITLE
Exported unnamed Node subclasses don't match anything in the node selector

### ISSUE THREAD

**Issue by vmedea** [godotengine/godot#86636]
### Tested versions

4.2-stable

### System information

N/A

### Issue description

With the following GDscript:
```
extends <SomeNode2D>

const GrowingRect = preload("growing_rect.gd")

@export var growing_rect: GrowingRect
```
where `growing_rect.gd` contains a class extending Node:
```
extends Node2D
...
```
The variable slot is successfully exported, however, it's not possible to select a node of this type in the editor inspector. The "Select a node" list remains empty, or when "Show All" is on, it's greyed out:
![image](https://github.com/godotengine/godot/assets/43048142/f938df59-0df5-4aa1-bf2f-e590f98a0c21)

This is inconvenient because it forces the use of global `class_name` for everything that's exported, cluttering the global namespace, instead of a scene being local and self-contained.

See also #77322, however, in contrast to that more general issue, the export works fine (by manually editing the scene file), it's just the editor that doesn't show matches.

### Steps to reproduce

- Open the attached minimal reproduction project
- Open "root.tscn" scene
- Select "PlayerController"
- In the inspector, remove the selected node for "GrowingRect" then go to the chooser and try to select the node again

### Minimal reproduction project (MRP)

[repro-anonnode.tar.gz](https://github.com/godotengine/godot/files/13798008/repro-anonnode.tar.gz)

**Comment by kleonc** [comment]
Already fixed by #82528 (included in v4.3.dev1.official [9d1cbab1c] and marked for cherry-picking into future `4.2.x`).

**Comment by vmedea** [comment]
Thank you, that's great news!
